src/type-layout.md

```diff
@@ -440,12 +440,56 @@ struct MyVariantD(MyEnumDiscriminant);
 
 > Note: `union`s with non-`Copy` fields are unstable, see [55149].
 
-For enums with fields, it is also possible to combining `repr(C)` and a
-primitive representation. This modifies the [`repr(C)`] by changing the
-representation of the discriminant enum to have the representation of the chosen
-primitive representation. So, if you chose the `u8` representation, then the
+#### Combining primitive representations of enums with fields and \#[repr(C)]
+
+For enums with fields, it is also possible to combine `repr(C)` and a
+primitive representation (e.g., `repr(C, u8)`). This modifies the [`repr(C)`] by
+changing the representation of the discriminant enum to have the representation
```

Comment:
Here: "changing the representation [...] to have the representation of".  But representations do not have representations.

Reply:
I updated the wording and added a little context to the example below, is it better this way ?

Reply:
Yes looks good. :)